server.js

```diff
@@ -1,23 +1,110 @@
 const express = require("express");
-const cors = require("cors");
+
 
 const app = express();
+const bodyParser = require("body-parser");
+app.use(bodyParser.urlencoded({
+    extended: true
+}));
+
+app.use(bodyParser.json())
+app.use(express.json())
+
 
-app.use(cors());
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
-  text: "Welcome to CYF chat system!",
-};
+  text: "Welcome to CYF chat system!"
+}
+
+const messages = [welcomeMessage]
+
+
+app.get('/', function(request, response) {
+  response.sendFile(__dirname + '/index.html');
+});
 
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+app.get('/messages', function(request, response) {
+  response.json(messages);
 });
 
-app.listen(process.env.PORT);
+app.get("/messages/search", (req, res)=>{
+  let text = req.query.text;
+  if(text !== undefined){
+      let  result = messages.filter(message =>message.text.toLowerCase().includes(text.toLowerCase()));
+if(result.length < 1){
+    res.status(400).json(`Please enter a valid search term!`)
+  }else{
+    res.json(result)
+  }
+  }
+})
```

Comment:
It would be good to handle the case where the user does not enter any search term - i.e. if req.query.text does not exist. At the moment the request will time out